### PR TITLE
Rework Asf, Riff video, test data and fix OSS-Fuzz issues

### PR DESCRIPTION
**Pull request by mohamedchebbii**
```
sudo ctest 
Test project /home/mohamed.chebbi/github/exiv2/build
    Start 1: bugfixTests
1/6 Test #1: bugfixTests ......................   Passed    5.94 sec
    Start 2: lensTests
2/6 Test #2: lensTests ........................   Passed    0.17 sec
    Start 3: tiffTests
3/6 Test #3: tiffTests ........................   Passed    0.12 sec
    Start 4: versionTests
4/6 Test #4: versionTests .....................   Passed    0.11 sec
    Start 5: regressionTests
5/6 Test #5: regressionTests ..................   Passed    3.04 sec
    Start 6: unitTests
6/6 Test #6: unitTests ........................   Passed    0.01 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) =   9.39 sec
```

### CODE REVIEW

**Comment by piponazo** [comment]
The changes look good, however the CI job "Linux-Ubuntu Quick Fuzz" is now failing with the new code. Could you try to run the fuzzer locally to try to fix the issues in the code?

The stack trace indicates that the faulty code is around this area:
```
#7 0x564966d0315c in operator delete(void*) (/home/runner/work/exiv2/exiv2/build/bin/fuzz-read-print-write+0x11815c) (BuildId: db0bbb4a7497f12dd44c77cc8d4d1a6bf2ff5699)
    #8 0x7f089f85cfe5 in Exiv2::RiffVideo::decodeBlock() (/home/runner/work/exiv2/exiv2/build/lib/libexiv2.so.0+0xfcbfe5) (BuildId: c5221ff9c3546e077be[32](https://github.com/Exiv2/exiv2/actions/runs/3859172205/jobs/6578457544#step:5:33)55b146c2597b530a76b)
    #9 0x7f089f85da10 in Exiv2::RiffVideo::tagDecoder(Exiv2::DataBuf&, unsigned long) (/home/runner/work/exiv2/exiv2/build/lib/libexiv2.so.0+0xfcca10) (BuildId: c5221ff9c[35](https://github.com/Exiv2/exiv2/actions/runs/3859172205/jobs/6578457544#step:5:36)46e077be3255b146c2597b530a76b)
    #10 0x7f089f85d[37](https://github.com/Exiv2/exiv2/actions/runs/3859172205/jobs/6578457544#step:5:38)f in Exiv2::RiffVideo::decodeBlock() (/home/runner/work/exiv2/exiv2/build/lib/libexiv2.so.0+0xfcc37f) (BuildId: c5221ff9c3546e077be3255b146c2597b530a76b)
    #11 0x7f089f85b4fb in Exiv2::RiffVideo::readMetadata() (/home/runner/work/exiv2/exiv2/build/lib/libexiv2.so.0+0xfca4fb) (BuildId: c5221ff9c3546e077be3255b146c2597b530a76b)
    #12 0x564966d0517b in LLVMFuzzerTestOneInput (/home/runner/work/exiv2/exiv2/build/bin/fuzz-read-print-write+0x11a17b) (BuildId: db0bbb4a7497f12dd44c77cc8d4d1a6bf2ff5699)
    #13 0x564966c2b483 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/home/runner/work/exiv2/exiv2/build/bin/fuzz-read-print-write+0x40483) (BuildId: db0bbb4a7497f12dd44c77cc8d4d1a6bf2ff5699)
```

File: src/asfvideo.cpp [hunk 1]
```diff
@@ -248,405 +238,273 @@ void AsfVideo::readMetadata() {
...
-    throw Error(ErrorCode::kerFailedToReadImageData);
-}  // AsfVideo::contentDescription
+}  // AsfVideo::extendedStreamProperties
 
 void AsfVideo::streamProperties() {
-  DataBuf buf(20);
-  byte guidBuf[GUI_SIZE];
-  int stream = 0;
+  byte streamTypedBuf[GUID];
+  io_->read(streamTypedBuf,GUID);
```
Comment (benmccann) [inline src/asfvideo.cpp:337]: clang-format-lint-action wants a space here
```suggestion
  io_->read(streamTypedBuf, GUID);
```

Comment (mohamedchebbii) [reply src/asfvideo.cpp]: Thank you @benmccann ,
Done here [b8f800e](https://github.com/Exiv2/exiv2/pull/2458/commits/b8f800ee4d784b179a934a2f38c7ddcd6762e1bc)

**Comment by mohamedchebbii** [comment]
Hello @piponazo,

**Comment by kevinbackhouse** [comment]
Thanks! I'll take a look later today.

**Comment by kevinbackhouse** [comment]
I'm going to fuzz it overnight on a 16-core machine. If that doesn't find any new issues, then I think we should change the default setting of `EXIV2_ENABLE_VIDEO` back on `ON` so that OSS-Fuzz starts fuzzing the video code again.

File: src/riffvideo.cpp [hunk 2]
```diff
@@ -527,660 +376,382 @@ void RiffVideo::readMetadata() {
...
+  EXV_DEBUG << "--> dwInitialFrames       = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwStreams             = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwSuggestedBufferSize = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwWidth               = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwHeight              = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved1           = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved2           = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved3           = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved4           = " << readDWORDTag(io_) << std::endl;
+#else
```
Comment (kevinbackhouse) [inline src/riffvideo.cpp:487]:
```suggestion
#endif
```

I don't think the code below this should be commented out when `EXIV2_DEBUG_MESSAGES` is on, should it?

If you agree with this suggestion, then there are several other places in this file that also need to be fixed.

**Comment by neheb** [comment]
Needs rebasing.

File: src/riffvideo.cpp (hunk 2 above)
Comment (mohamedchebbii) [reply src/riffvideo.cpp]: Thank you @kevinbackhouse ,
Done  here [db14131](https://github.com/Exiv2/exiv2/pull/2458/commits/db1413126237bc8cc81abd6a583e4e9782f3e2aa)

**Comment by mohamedchebbii** [comment]
> Needs rebasing.

Done here [3847a09](https://github.com/Exiv2/exiv2/pull/2458/commits/3847a0935957eca5d707e277596489ffcf6c0a0f)

**Comment by kevinbackhouse** [comment]
Seems like there are still some issues. Here's a poc that the fuzzer found: 

https://user-images.githubusercontent.com/4358136/216977045-0ea60ce7-ca64-4163-ac6a-2e9d7f48a3cc.mp4

You need to build with ASAN to reproduce:

```bash
cmake -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_ENABLE_VIDEO=ON -DEXIV2_TEAM_USE_SANITIZERS=ON ..
make
./bin/exiv2 poc.mp4 
```

**Comment by kevinbackhouse** [comment]
Looks like I need top gzip the poc: 
[poc.mp4.gz](https://github.com/Exiv2/exiv2/files/10663285/poc.mp4.gz)

**Comment by mohamedchebbii** [comment]
> Looks like I need to gzip the poc: [poc.mp4.gz](https://github.com/Exiv2/exiv2/files/10663285/poc.mp4.gz)

Thank you @kevinbackhouse , done here [579cae8]

**Comment by benmccann** [comment]
Maybe it would be valuable to merge this since it's quite an improvement and then handle any additional fuzzing issues discovered in a follow-up?

**Review by neheb — DISMISSED** [review]

**Comment by neheb** [comment]
This branch cannot be rebased due to conflicts

Needs a rebase. Again.

**Review by neheb — APPROVED** [review]

**Comment by neheb** [comment]
This branch cannot be rebased due to conflicts

please use

 ```
git fetch origin
git rebase origin/main
git push --force
```

I can't merge.

**Comment by neheb** [comment]
ehhh merge commit seems to work.

**Pull request merged**